src/main/knn/KnnGraphTester.java

```diff
@@ -571,72 +603,60 @@ private void testSearch(Path indexPath, Path queryPath, Path outputPath, int[][]
             } else {
               float[] target = targetReader.next();
               if (prefilter) {
-                results[i] = doKnnVectorQuery(searcher, KNN_FIELD, target, topK, fanout, bitSetQuery);
+                results[i] = doKnnVectorQuery(searcher, KNN_FIELD, target, topK, fanout, bitSetQuery, parentJoin);
               } else {
                 results[i] =
                   doKnnVectorQuery(
-                                   searcher, KNN_FIELD, target, (int) (topK / selectivity), fanout, null);
+                    searcher, KNN_FIELD, target, (int) (topK / selectivity), fanout, null, parentJoin);
+              }
+              if (prefilter == false && matchDocs != null) {
+                results[i].scoreDocs =
+                  Arrays.stream(results[i].scoreDocs)
+                    .filter(scoreDoc -> matchDocs.get(scoreDoc.doc))
+                    .toArray(ScoreDoc[]::new);
               }
-            }
-            if (prefilter == false && matchDocs != null) {
```

Comment:
Whoa, was this pre-existing bug (wrong nesting)?

Reply:
I think it was an indentation issue. The nesting was fine.